File: Mobile App/Apollo Deliver/lib/Models/Store.dart
import 'package:apollodeliver/Models/Admin.dart';
import 'package:apollodeliver/Models/Product.dart';
import 'package:apollodeliver/Models/StoreCategory.dart';

class Store {
  int? id;
  String? name;
  String? logo;
  String? homeImage;
  String? dealsImage;
  String? dealsSquareImage;
  String? interactiveImage;
  Admin? admin;
  List<StoreCategory>? storeCategoryList;
  List<Product>? productList;

  Store({
    this.id,
    this.name,
    this.logo,
    this.homeImage,
    this.dealsImage,
    this.dealsSquareImage,
    this.interactiveImage,
    this.admin,
    this.storeCategoryList,
    this.productList,
  });

  factory Store.fromJson(Map<String, dynamic> json) {
    return Store(
      id: json['id'],
      name: json['name'],
      logo: json['logo'],
      homeImage: json['homeImage'],
      dealsImage: json['dealsImage'],
      dealsSquareImage: json['dealsSquareImage'],
      interactiveImage: json['interactiveImage'],
      admin: json['admin'] != null ? Admin.fromJson(json['admin']) : null,
      storeCategoryList: (json['storeCategoryList'] as List<dynamic>?)
          ?.map((item) => StoreCategory.fromJson(item))
          .toList(),
      productList: (json['productList'] as List<dynamic>?)
          ?.map((item) => Product.fromJson(item))
          .toList(),
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'name': name,
    'logo': logo,
    'homeImage': homeImage,
    'dealsImage': dealsImage,
    'dealsSquareImage': dealsSquareImage,
    'interactiveImage': interactiveImage,
    'admin': admin?.toJson(),
    'storeCategoryList': storeCategoryList?.map((item) => item.toJson()).toList(),
    'productList': productList?.map((item) => item.toJson()).toList(),
  };
}

File: API/apollo_api_services/src/main/java/com/apollo/entity/Store.java
package com.apollo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.util.List;

@Entity
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Table(name="store")

public class Store {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String logo;
    private String homeImage;
    private String dealsImage;
    private String dealsSquareImage;
    private String interactiveImage;

    @ManyToOne
    @JoinColumn(name="admin_id")
    private Admin admin;

    @OneToMany(mappedBy = "store")
    @JsonIgnore
    private List<StoreCategory> storeCategoryList;

    @OneToMany(mappedBy = "store")
    @JsonIgnore
    private List<Product> productlist;

}
